game/src/Engine.cpp

```diff
@@ -33,6 +36,14 @@ void Engine::startGameLoop(){
         std::vector<std::tuple<SDL_Rect, int> > bullets;
         auto eventStatus = SDL_PollEvent(&event);
 
+        if (event.key.keysym.sym == SDLK_ESCAPE) {
+            value = true;
+        }
+        while (value) {
```

Comment:
this is where you will need the ifs and the continue. you will check for the escape event and if you see it enter the pause state